tests/test_datetime/test_datetime_features.py

```diff
@@ -148,7 +159,11 @@ def test_raises_error_when_nan_in_transform_df(df_datetime):
     transformer.fit(df_datetime)
     # dataset containing nans
     with pytest.raises(ValueError):
-        DatetimeFeatures().transform(dates_nan)
+        transformer.transform(dates_nan)
+    transformer = DatetimeFeatures(variables="index")
```

Comment:
I see, here is when the problem you raised would come up. If the user has a variable called "index", but we are now using this string to indicate the df index, then they would not be able to extract features from this variable.

I think this is very unlikely, but the user still could pass ["index"] and then, they could extract features from the column called index. And if they pass "index" as a string, then, the class would use the index of the df. And we could make this clear in the docstrings. 

I think, as a first release, this could be enough, and then we hear user feedback. Unless you think otherwise?

Reply:
Yea no I agree. If anything if someone comes up with a dataset where a column not only is named "index" but contains datetime info (??) it's probably their fault in terms of bad naming practices and even lack of common sense haha